notebooks/228-clip-zero-shot-image-classification/228-clip-zero-shot-image-classification.ipynb

```diff
@@ -0,0 +1,281 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!python -m transformers.onnx --model openai/clip-vit-base-patch16 clip-vit-base-patch16"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from openvino.runtime import serialize\n",
+    "from openvino.tools import mo\n",
+    "\n",
+    "ov_model = mo.convert_model('clip-vit-base-patch16/model.onnx', compress_to_fp16=True)\n",
+    "serialize(ov_model, 'clip-vit-base-patch16/model.xml')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Run OpenVINO model\n",
+    "\n",
+    "The steps for making predictions with OpenVINO CLIP model are the same like for PyTorch model. Let's check model result."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "from scipy.special import softmax\n",
+    "from openvino.runtime import Core\n",
+    "\n",
+    "# create OpenVINO core object instance\n",
+    "core = Core()\n",
+    "# compile model for loading on device\n",
+    "compiled_model = core.compile_model(ov_model)\n",
+    "# obtain output tensor for getting predictions\n",
+    "logits_per_image_out = compiled_model.output(0)\n",
+    "# run inference on preprocessed data and get image-text similarity score\n",
+    "ov_logits_per_image = compiled_model(dict(inputs))[logits_per_image_out]\n",
+    "# perform softmax on score\n",
+    "probs = softmax(ov_logits_per_image, axis=1)\n",
+    "# visualize prediction\n",
+    "visualize_result(image, input_labels, probs[0])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Great! Looks like we got the same result."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, is your turn. You can provide own image and comma-separated list of labels for zero-shot classification"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import ipywidgets as widgets\n",
+    "style = {'description_width': 'initial'}\n",
+    "\n",
+    "image_widget = widgets.FileUpload(\n",
+    "    accept='',\n",
+    "    multiple=False,\n",
+    "    description='Upload image',\n",
+    "    style=style\n",
+    ")\n",
+    "\n",
+    "labels_widget = widgets.Textarea(\n",
+    "    value='cat,dog,bird',\n",
```

Comment:
```suggestion
    "    value='cat,dog,bird',\n #Assing your labels here",
```

Reply:
Not sure that we need this comment, it is default values used as example, I do not think that users should modify them. Instead user should type his labels to text field ui element

Reply:
I think we need to improve the instructions. This is just an example.

Reply:
My initial idea to have something like that https://huggingface.co/spaces/ShivamShrirao/CLIP-Zero-Shot-Classifier

But I agree that with widgets, it looks not as impressive and self-descriptive like with gradio, so I'll provide additional instruction

Reply:
added instruction for providing data, please take a look